aesara/link/numba/dispatch/tensor_basic.py

```diff
@@ -198,11 +198,14 @@ def makevector({", ".join(input_names)}):
 
 @numba_funcify.register(Rebroadcast)
 def numba_funcify_Rebroadcast(op, **kwargs):
-    op_axis = tuple(op.axis.items())
+    # GH issue https://github.com/numba/numba/issues/8215
+    op_axis = tuple((axis, int(value)) for axis, value in op.axis.items())
```

Comment:
If this is homogenizing the axis type as `((int, int), ^ N )` is it necessary to version the loop body with the unroller? Is there a chance axis can be `None` perhaps, in which case this seems valid?

Reply:
> version the loop body with the unroller

I don't understand what you mean here.

Reply:
> I see, thanks for explaining!

No problem. I wasn't sure in the case above whether you'll sometimes have e.g. a `None` for `axis`, or perhaps you need to preserve the literal nature of values captured scope of `op_axis`, but if you don't, I think you might be able to get away with standard iteration, which will get you a faster loop (as there's no switch table in the body).

Reply:
I removed the `literal_unroll` for now.

Reply:
Sounds good. For performance, if you can invent a way to make a loop type stable, that's usually a good approach as it means there's no need for a switch table and duplicated loop bodies. LLVM can often "see" through the "switch table" pattern but it's better,  if possible, to avoid creating the risk of it not being able to optimising it away by simply avoiding it in the program source.

Thanks for working through this issue. The Numba folks will take a look at the segfault(s) next week!

Reply:
@stuartarchibald Thank you for the super fast help!

Reply:
> @stuartarchibald Thank you for the super fast help!

No problem! Thanks for using Numba!